### PR TITLE
Rename Parameter; Add Code Coverage to Test

### LINKED ISSUE #4248: Fix PSScriptAnalyzerRule - PSAvoidAssignmentToAutomaticVariable
### Summary of new feature

Currently Read-DbaTraceFile uses the automatic variable `$Error` as a parameter on the cmdlet. This can cause issues as `$Error` is an automatic variable that is set as Read Only. 

### Proposed technical details (if applicable)

Replace the current parameter `-Error` with a parameter like `-ErrorID` and possibly include a `Test-DbaDeprecation` until V1.0

### PR DESCRIPTION
**Pull request by joshcorr**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #4248 )
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
Remove the Parameter `-Error` from `Read-DbaTraceFile`
Replace with `-ErrorID` (This is not the value of the resultant table, but seemed the most logical name)
Add Code Coverage to existing Pester Test

### Approach

Replaced Instances of `$Error` with `$ErrorID`
Gave a little more explanation in the parameter description

### Commands to test

` Invoke-ScriptAnalyzer .\functions\Read-DbaTraceFile.ps1`
`Invoke-Pester .\Tests\Read-DbaTraceFile.Tests.ps1`

### CODE REVIEW

File: functions/Read-DbaTraceFile.ps1 [hunk 1]
```diff
@@ -168,7 +168,7 @@ function Read-DbaTraceFile {
         [int[]]$Spid,
         [string[]]$EventClass,
         [string[]]$ObjectType,
-        [int[]]$Error,
+        [int[]]$ErrorID,
```
Comment (potatoqualitee) [inline functions/Read-DbaTraceFile.ps1]: thanks so much! this should be $ErrorId, however, then we can merge 
```suggestion
        [int[]]$ErrorId,
```

Comment (potatoqualitee) [reply functions/Read-DbaTraceFile.ps1]: hey cool suggestion!

Comment (joshcorr) [reply functions/Read-DbaTraceFile.ps1]: That is a cool feature makes code reviews easier. Thanks for all the positive feed back on these PRs, You are doing a great job heading up this project!

Comment (potatoqualitee) [reply functions/Read-DbaTraceFile.ps1]: Thanks so much 🤗 I'm always happy to see these PRs coming in. I think it keeps the momentum going. Finally 1.0 seems possible!

**Pull request merged**